### PR TITLE
Add ProxyV2 resource map

### PR DESCRIPTION
**Pull request by msukruth**

### CODE REVIEW

File: retrieval/resource_map.go [hunk 1]
```diff
@@ -117,10 +117,25 @@ var ProxyResourceMap = ResourceMap{
 	},
 }
 
+var AggregatedProxyResourceMap = ResourceMap{
```
Comment (StevenYCChou) [inline retrieval/resource_map.go:120]:
```suggestion
var ProxyV2ResourceMap = ResourceMap{
```

Call it ProxyV2 would be reasonable for the user as the type is ProxyV2.

File: retrieval/resource_map.go [hunk 2]
```diff
@@ -117,10 +117,25 @@ var ProxyResourceMap = ResourceMap{
...
+	Type:       "apigee.googleapis.com/ProxyV2",
+	MatchLabel: "__meta_kubernetes_pod_label_type_aggregate_proxy",
```
Comment (StevenYCChou) [inline retrieval/resource_map.go:122]: Is there any prior agreement that the pod label is `aggregate_proxy`? If yes, I am fine with it;

but if not, I think the pod_label as `proxy_v2` and the match label as `__meta_kubernetes_pod_label_type_proxy_v2` may be reasonable from the user perspective, not from how it is implemented.

What do you think?

File: retrieval/resource_map.go (hunk 1 above)
Comment (msukruth) [reply retrieval/resource_map.go]: Done

File: retrieval/resource_map.go (hunk 2 above)
Comment (msukruth) [reply retrieval/resource_map.go]: Sounds Good. Updated the CL with the same.

**Comment by StevenYCChou** [comment]
> Add the expected order to the test function `TestResourceMappingsOrder`. An example can be found: https://github.com/Stackdriver/stackdriver-prometheus-sidecar/pull/189/files#diff-df24886d8b2bea2712e71ffcae5f1db7R231-R236

@msukruth can you also update this test function?

**Comment by StevenYCChou** [comment]
Travis build fails caused by coverall; the actual test are done in https://travis-ci.com/github/Stackdriver/stackdriver-prometheus-sidecar/builds/171465978 as 
```
ok  	github.com/Stackdriver/stackdriver-prometheus-sidecar/retrieval	2.293s	coverage: 89.0% of statements
```

**Pull request merged**